src/compiler/parser.ts

```diff
@@ -1824,14 +1861,13 @@ module ts {
         function parseParameter(): ParameterDeclaration {
             let node = <ParameterDeclaration>createNode(SyntaxKind.Parameter);
             node.decorators = parseDecorators();
-            setModifiers(node, parseModifiers());
             node.dotDotDotToken = parseOptionalToken(SyntaxKind.DotDotDotToken);
+            setModifiers(node, parseModifiers());
 
-            // SingleNameBinding[Yield,GeneratorParameter] : See 13.2.3
-            //      [+GeneratorParameter]BindingIdentifier[Yield]Initializer[In]opt
-            //      [~GeneratorParameter]BindingIdentifier[?Yield]Initializer[In, ?Yield]opt
+            // FormalParameter[Yield,GeneratorParameter,Await,AsyncParameter] : (Modified) See 14.1
+            //      BindingElement[?Yield,?GeneratorParameter,?Await,?AsyncParameter]
 
-            node.name = inGeneratorParameterContext() ? doInYieldContext(parseIdentifierOrPattern) : parseIdentifierOrPattern();

```

Comment:
This isn't needed per the grammar. From the ES6 grammar:
```
SingleNameBinding[Yield,GeneratorParameter] : 
     [+GeneratorParameter] BindingIdentifier[Yield] Initializer[In]opt
     [~GeneratorParameter] BindingIdentifier[?Yield] Initializer[In,?Yield]opt
```

The *BindingIdentifier* will always be parsed with the ambient `[Yield]` (as `[Yield]` is always set when `[GeneratorParameter]` is set). The above two productions really only differ when parsing the *Initializer* non-terminal.

The same is the case when parsing a `BindingElement`:
```
BindingElement[Yield,GeneratorParameter] :
     [+GeneratorParameter] BindingPattern[?Yield,GeneratorParameter] Initializer[In]opt
     [~GeneratorParameter] BindingPattern[?Yield] Initializer[In,?Yield]opt
```

Really, the *SingleNameBinding* productions in the ES6 grammar could just read the following and have the same effect:
```
SingleNameBinding[Yield,GeneratorParameter] : 
     [+GeneratorParameter] BindingIdentifier[?Yield] Initializer[In]opt
     [~GeneratorParameter] BindingIdentifier[?Yield] Initializer[In,?Yield]opt
```